src/index.js

```diff
@@ -1,23 +1,40 @@
 import Todo from './todo.js';
-import checkBoxEvent from './checkbox.js';
 import './style.css';
 
 const TODO_LIST_KEY = 'TODO_LIST_KEY';
+const template = document.querySelector('#list-item-template');
+const todoListContainer = document.querySelector('#list');
+const form = document.querySelector('.form');
+const inputField = document.querySelector('#todo-input');
+const clearAllCompleted = document.querySelector('.clear-all-completed');
 
 const loadList = () => {
   const dataInStringFormat = localStorage.getItem(TODO_LIST_KEY);
   return JSON.parse(dataInStringFormat) || [];
 };
 
-const todoList = loadList();
-
 const saveList = () => {
   localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
 };
 
-const todoListContainer = document.querySelector('#list');
-const form = document.querySelector('.form');
-const inputField = document.querySelector('#todo-input');
+const renderTodo = (todo) => {
+  const templateClone = template.content.cloneNode(true);
+  const taskContent = templateClone.querySelector('[data-list-item-text]');
+  taskContent.innerText = todo.description;
+  const checkBox = templateClone.querySelector('[data-list-item-checkbox]');
+  checkBox.checked = todo.completed;
+  checkBox.addEventListener('change', () => {
+    todo.completed = checkBox.checked;
+    // saveList();
+    localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+  });
+  const listItem = templateClone.querySelector('.list-item');
+  listItem.dataset.todoIndex = todo.index;
```

Comment:
- [ ] The property index should be set to the value of the new array length. As you can see from the screenshot below the last index should be 3 and not 2
![image](https://user-images.githubusercontent.com/68167810/130330046-adc42b57-a25c-444e-9e24-67bd2b62aa77.png)

Reply:
Alright